lib/src/components/NxTabs/__tests__/NxTabs.test.tsx

```diff
@@ -5,103 +5,76 @@
+    onTabSelect: () => {},
+    children: (
+      <NxTabList>
+        <NxTab>Tabs 0</NxTab>
+      </NxTabList>
+    )
+  };
 
-    expect(firstTabs).not.toHaveProp('id', secondTabs.prop('id'));
-  });
+  const quickRender = rtlRender<NxTabsProps>(NxTabs, minimumProps);
+  const renderEl = rtlRenderElement<NxTabsProps>(NxTabs, minimumProps);
 
-  it('renders with the correct classnames', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1} className="test">
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('merges any passed in className', function() {
+    const tabsWithAddedClassName = renderEl({ activeTab: -1, className: 'foo' });
+    const tabs = renderEl({ activeTab: -1 })!;
+
+    expect(tabsWithAddedClassName).toHaveClass('foo');
 
-    expect(component).toHaveClassName('nx-tabs test');
+    for (const cls of Array.from(tabs.classList)) {
+      expect(tabsWithAddedClassName).toHaveClass(cls);
+    }
   });
 
-  it('renders a generated id', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1}>
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('renders with a unique id each time', function() {
+    const firstTabs = renderEl({ activeTab: 0 })!;
+    const secondTabs = renderEl({ activeTab: 0 })!;
 
-    expect(component.prop('id')).toMatch(/^nx-tabs-\d+$/);
+    expect(firstTabs.id).not.toEqual(secondTabs.id);
   });
 
-  it('renders using as specific id', function () {
-    const component = shallow(
-      <NxTabs onTabSelect={() => {}} activeTab={-1} id="my-tabs">
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('renders a generated id', function() {
+    const component = renderEl({ activeTab: -1 })!;
+    expect(component.id).toMatch(/^nx-tabs-\d+$/);
+  });
 
-    expect(component.prop('id')).toMatch('my-tabs');
+  it('renders using as specific id', function() {
+    const commponent = renderEl({
+      id: 'my-tabs',
+      activeTab: -1
+    })!;
+    expect(commponent.id).toMatch('my-tabs');
   });
 
-  it('renders no tab contents when none are active', function () {
-    const component = mount(
-      <NxTabs onTabSelect={() => {}} activeTab={-1}>
-        <NxTabList>
-          <NxTab>Tab 0</NxTab>
-        </NxTabList>
-        <NxTabPanel>Content 0</NxTabPanel>
-      </NxTabs>
-    );
+  it('renders no tab contents when none are active', function() {
+    const { getByRole, queryByRole } = quickRender({ activeTab: -1 });
 
-    expect(component.find(NxTab)).not.toHaveClassName('active');
-    expect(component.find(NxTabPanel)).toBeEmptyRender();
+    expect(getByRole('tab').classList.contains('active')).toBeFalsy();
```

Comment:
Check `aria-selected` rather than this class

Reply:
> Check aria-selected rather than this class

One way to do that is with the extra parameters to `queryByRole`. https://testing-library.com/docs/queries/byrole/

Reply:
> > Check aria-selected rather than this class
> 
> One way to do that is with the extra parameters to `queryByRole`. https://testing-library.com/docs/queries/byrole/

Ooo thanks! updated here: 9ee3fecf501160880785f087d981f677b0a2c974